ch3/exercise3-11/comma.go

```diff
@@ -16,35 +16,42 @@ func main() {
 
 func comma(s string) string {
 
-	match, _ := regexp.MatchString("\\.", s)
-	signMatch, _ := regexp.MatchString("[+-]", s)
+	pointMatch, err := regexp.MatchString("\\.", s)
+	if err != nil {
+		return s + ": " + err.Error()
+	}
+	signMatch, err := regexp.MatchString("[+-]", s)
+	if err != nil {
+		return s + ": " + err.Error()
+	}
```

Comment:
`s` is irrelevant to the error, and potentially misleading.  `MatchString` will only fail when compiling the pattern fails:

https://golang.org/src/regexp/regexp.go?s=16946:17014#L538

So the pattern would be the relevant debugging output.  This is partially why I suggested you split compiling and applying the regex.